.github/workflows/publish-test-result.yml

```diff
@@ -0,0 +1,58 @@
+name: Run unit test
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run unit tests and capture results
+        id: unit-test-results
+        run: |
+          yarn test --json --outputFile=test-results.json
+
+      - name: Publish test results
+        uses: actions/github-script@v6
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const {repo, owner} = context.repo
+            const fs = require('fs');
+            const results = JSON.parse(fs.readFileSync('test-results.json', 'utf8'));
```

Comment:
Could we get the code coverage of a test file from the result? If yes, please show it in the Test Results like this format.
[Test format](https://user-images.githubusercontent.com/62632321/218383185-0581c50c-2c18-4429-8b93-10d2a2325714.png)